src/forms/ToggleSwitch.jsx

```diff
@@ -0,0 +1,127 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+
+import { C_TEXT_PRIMARY, C_TEXT_SECONDARY, C_TEXT_HINT } from 'swarm-constants/dist/js/colorConstants.js';
+
+import Button from './Button';
+import Icon from '../media/Icon';
+import withToggleControl from '../utils/WithToggleControl';
+
+export const TOGGLE_SWITCH_CLASS = 'toggleSwitch';
+export const TOGGLE_SWITCH_ACTIVE_CLASS = `${TOGGLE_SWITCH_CLASS}--active`;
+export const TOGGLE_SWITCH_DISABLED_CLASS = `${TOGGLE_SWITCH_CLASS}--disabled`;
+
+export const KNOB_CLASS = `${TOGGLE_SWITCH_CLASS}-knob`;
+export const KNOB_ACTIVE_CLASS = `${KNOB_CLASS}--active`;
+export const KNOB_DISABLED_CLASS = `${KNOB_CLASS}--disabled`;
+
+export const LABEL_CLASS = `${TOGGLE_SWITCH_CLASS}-label`;
+
+/**
+ * @module ToggleSwitch
+ */
+class ToggleSwitch extends React.Component {
+	constructor (props) {
+		super(props);
+
+		this.toggle = this.toggle.bind(this);
+	}
+
+	toggle() {
+		!this.props.disabled && this.props.toggleActive();
+	}
+
+	componentWillReceiveProps(nextProps) {
```

Comment:
Now I'm wondering if putting this in `componentDidUpdate` -- it still receives `prevProps`. Also, either lifecycle function will get call many times so I think you should be comparing the current and past state of `this.props.isActive` and only calling the appropriate callback when isActive has actually changed. It will prevent a callback from getting called multiple times.